### ISSUE TITLE
[Bug]: Edge lambda rendering not working for root page

### ISSUE THREAD

**Issue by ShivamJoker** [lukeshay/astro-aws#47]
### 💻

- [ ] Would you like to work on a fix?

### Input code

I am using the SSR example from astro repo - https://github.com/withastro/astro/tree/main/examples/ssr

```js
const astroWebsite = new AstroAWS(this, "AstroAWSConstruct", {
  output: "edge",
  websiteDir: "../website",
  cdk: {
    cloudfrontDistribution: {
      defaultBehavior: {
        cachePolicy: cachePolicy,
      },
    },
  },
});
```

### Current and expected behavior

The `/api` endpoint is working fine, however the rendering of homepage is giving me **503 ERROR The request could not be satisfied.**

![image](https://user-images.githubusercontent.com/23727670/222208370-afc1565d-c045-4ab3-86a7-04a095221b67.png)

### Environment

"@astro-aws/constructs": "^0.3.0",
"aws-cdk-lib": "2.66.1",
"constructs": "^10.0.0",
"@astro-aws/adapter": "^0.3.0",

### Possible solution

No idea. Where can I see the rendering log?
Do I need to enable Cloudfront logging?

**Comment by lukeshay** [comment]
I will take a look. If it is reaching the lambda, you can see the logs in Cloudwatch.

**Comment by ShivamJoker** [comment]
@lukeshay I tried looking in the Cloudwatch but there was no logs for the edge lambda.
Can it be because of permissions?

**Comment by ShivamJoker** [comment]
Just the usual

```js
import { defineConfig } from "astro/config";
import svelte from "@astrojs/svelte";
import aws from "@astro-aws/adapter";

// https://astro.build/config
export default defineConfig({
  output: "server",
  adapter: aws(),
  integrations: [svelte()],
});
```

**Comment by JohannesKlauss** [comment]
Having the same issue. Any updates on that? It seems that like in General GET Requests are failing and every other type of HTTP request succeeds. If I test my lambda with `uri: '/'`, but make a POST request out of it it works fine. I'll dig around some more

**Comment by JohannesKlauss** [comment]
It's a bug in Astro, I pushed a PR for the fix here: https://github.com/withastro/astro/pull/6688

In the meantime you can fix it by manually adding an additional null check in the output in NodeApp.render:

```js
var NodeApp = class extends App {
  match(req, opts = {}) {
    return super.match(req instanceof Request ? req : createRequestFromNodeRequest(req), opts);
  }
  render(req, routeData) {
    if (typeof req.body === "string" && req.body.length > 0) {
      return super.render(
        req instanceof Request ? req : createRequestFromNodeRequest(req, Buffer.from(req.body)),
        routeData
      );
    }
    // Add null check to this line.
    if (typeof req.body === "object" && req.body !== null && Object.keys(req.body).length > 0) {
      return super.render(
        req instanceof Request ? req : createRequestFromNodeRequest(req, Buffer.from(JSON.stringify(req.body))),
        routeData
      );
    }

  // ....
```

**Comment by JohannesKlauss** [comment]
It's fixed in Astro 2.1.8
